src/sema/expression/member_access.rs

```diff
@@ -242,6 +244,27 @@ pub(super) fn member_access(
                     }
                     ArrayLength::AnyFixed => unreachable!(),
                 };
+            } else if matches!(*elem_ty, Type::Struct(StructType::AccountInfo))
+                && context.function_no.is_some()
+            {
+                return if ns.functions[context.function_no.unwrap()]
+                    .solana_accounts
+                    .borrow()
+                    .contains_key(&id.name)
+                {
+                    Ok(Expression::NamedSubscript {
+                        loc: *loc,
+                        ty: Type::Ref(Box::new(Type::Struct(StructType::AccountInfo))),
+                        array: Box::new(expr),
+                        name: id.name.clone(),
+                    })
+                } else {
+                    diagnostics.push(Diagnostic::error(
+                        id.loc,
+                        "unrecognized account".to_string(),
+                    ));
+                    Err(())
```

Comment:
Returning `Err(())` means we stop resolving the body of this function. Can we return the `Ok(Expression..)` even though it is not valid? As an error diagnostic has been generated, codegen should never be entered for the generated ast.

Reply:
Should we have an `Expression::Unresolved`?

Reply:
I tried returning a fake expression, but now we stop resolving the file.

Reply:
> I tried returning a fake expression, but now we stop resolving the file.

This is not what happens. I analyzed the issue and if I do not return an error, the diagnostics are not propagated to Namespace and they are not printed.

Reply:
If I write `assert(tx.accounts.unrecognized_account == address(this))` and I don't return `Err()`, the diagnostic is ignored here:

https://github.com/hyperledger/solang/blob/a1913b695374e869ee2714bd0c3dd7886545bcf8/src/sema/builtin.rs#L962-L975